api/envoy/extensions/filters/udp/udp_proxy/v3/udp_proxy.proto

```diff
@@ -18,10 +18,25 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // [#extension: envoy.filters.udp_listener.udp_proxy]
 
 // Configuration for the UDP proxy filter.
+// [#next-free-field: 6]
 message UdpProxyConfig {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.config.filter.udp.udp_proxy.v2alpha.UdpProxyConfig";
 
+  // Specifies the udp hash policy.
+  // The packets can be routed by hash policy.
+  message HashPolicy {
+    // The source IP will be used to compute the hash used by hash-based load balancing algorithms.
+    message SourceIp {
+    }
+
+    oneof policy_specifier {
+      option (validate.required) = true;
+
+      SourceIp source_ip = 1;
```

Comment:
I think you can use a bool here in which you verify using PGV that the only valid value is true. See other examples.

Reply:
Thanks!! That is more simple and clarified.